core/src/filesystem/media/analyze_media_job/mod.rs

```diff
@@ -154,85 +190,17 @@ impl JobExt for AnalyzeMediaJob {
 		let mut output = Self::Output::default();
 
 		match task {
-			AnalyzeMediaTask::AnalyzeImage(id) => {
-				// Get media by id from the database
-				let media_item: Media = ctx
-					.db
-					.media()
-					.find_unique(media::id::equals(id.clone()))
-					.with(media::metadata::fetch())
-					.exec()
-					.await
-					.map_err(|e: prisma_client_rust::QueryError| {
-						JobError::TaskFailed(e.to_string())
-					})?
-					.ok_or_else(|| {
-						JobError::TaskFailed(format!(
-							"Unable to find media item with id: {}",
-							id
-						))
-					})?
-					.into();
-
-				let path = media_item.path;
-				let page_count = get_page_count(&path, &ctx.config)?;
-				output.images_analyzed += 1;
-
-				// Check if a metadata update is neded
-				if let Some(metadata) = media_item.metadata {
-					// Great, there's already metadata!
-					// Check if the value matches the currently recorded one, update if not.
-					if let Some(meta_page_count) = metadata.page_count {
-						if meta_page_count != page_count {
-							ctx.db
-								.media_metadata()
-								.update(
-									media_metadata::media_id::equals(media_item.id),
-									vec![media_metadata::page_count::set(Some(
-										page_count,
-									))],
-								)
-								.exec()
-								.await?;
-							output.media_updated += 1;
-						}
-					} else {
-						// Page count was `None` so we update it.
-						ctx.db
-							.media_metadata()
-							.update(
-								media_metadata::id::equals(media_item.id),
-								vec![media_metadata::page_count::set(Some(page_count))],
-							)
-							.exec()
-							.await?;
-						output.media_updated += 1;
-					}
-				} else {
-					// Metadata doesn't exist, create it
-					let new_metadata = ctx
-						.db
-						.media_metadata()
-						.create(vec![
-							media_metadata::id::set(media_item.id.clone()),
-							media_metadata::page_count::set(Some(page_count)),
-						])
-						.exec()
-						.await?;
-
-					// And link it to the media item
-					ctx.db
-						.media()
-						.update(
-							media::id::equals(media_item.id),
-							vec![media::metadata::connect(media_metadata::id::equals(
-								new_metadata.id,
-							))],
-						)
-						.exec()
-						.await?;
-					output.media_updated += 1;
-				}
+			AnalyzeMediaTask::UpdatePageCount(id) => {
+				task_page_count::execute(id, ctx, &mut output).await?
+			},
+			AnalyzeMediaTask::AnalyzePageDimensions(id) => {
+				task_analyze_dimensions::execute(id, ctx, &mut output).await?
+			},
+			AnalyzeMediaTask::FullAnalysis(id) => {
+				// First page count needs to be updated
+				task_page_count::execute(id.clone(), ctx, &mut output).await?;
+				// Then we can do the dimensions analysis
+				task_analyze_dimensions::execute(id, ctx, &mut output).await?
```

Comment:
Yeah that's a good point - I should fix these tasks to use the same buffer instead of doing it twice.

I think it seems likely that this will be fixed in a future change that establishes a more pipeline-based method for media analysis. I'll add a todo for now flagging this inefficiency.